### ISSUE TITLE
how to install on nvidia jetson?

### ISSUE THREAD

**Issue by AndreV84** [intel-isl/Open3D#2134]
how to install it with cuda support? with librealsense  support?
```

[ 15%] Completed 'ext_turbojpeg'
[ 15%] Built target ext_turbojpeg
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[ 15%] No update step for 'ext_mkl'
[ 16%] No patch step for 'ext_mkl'
[ 16%] No configure step for 'ext_mkl'
[ 16%] Performing build step for 'ext_mkl'
Extracting static libs...
Merging static libs...
Cleaning up *.o files...
[ 17%] Performing install step for 'ext_mkl'
[ 17%] Completed 'ext_mkl'
[ 17%] Built target ext_mkl
Makefile:148: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by yxlao** [comment]
There are 3 questions:
- MKL: MKL is x86 only, you've bring up a good point, we'll consider adding BLAS+LAPACK support for Linux as an MKL alternative
- CUDA: the CUDA part should work with ARM (but of course, the build is currently blocked by MKL)
- librealsense : we haven't tested librealsense on ARM, please feel free to reach out to the librealsense community and let us know your findings

For the time being, if you like to try building on ARM, I'd recommend using one of your released version (not master) and disable the sensor component.

**Comment by nanotuxi** [comment]
Got open3d running on the Jetson Nano with cuda supprt today. Sensor component works fine, too. Going to write a short summary or howto tomorrow. 
Thanks.

**Comment by nanotuxi** [comment]
Here is a patch for the ndeds repo with a very detailed explanation on how to get Open3D running on a Jetson Nano
https://github.com/nanotuxi/jetsonNanoPatches

**Comment by AndreV84** [comment]
@nanotuxi 
Thank you or the update!

**Comment by nanotuxi** [comment]
@AndreV84 
De nada. From another thread I guess you have a L515? Did you get that camera working on the jetson nano, too?

**Comment by AndreV84** [comment]
yes L515 works on jetson

**Comment by nanotuxi** [comment]
@AndreV84 
Thanks.

**Comment by AndreV84** [comment]
@nanotuxi by some reason it appears that for python the package builds wth name open3d_pybind, that couldn't be imported as open3d but as open3d_pybind:
lib/RelWithDebInfo/Python/open3d_pybind.cpython-36m-aarch64-linux-gnu.so
Thus question how to build pythin wheel for open3d with your proposed approach? 
Thanks

**Comment by AndreV84** [comment]
`/o3d/git_src/Open3D/theNded/Open3D/build$ make install-pip-package
[ 70%] Built target EncodeShader
[ 70%] Built target ShaderFileTarget
[ 70%] Built target 3rdparty_tinyfiledialogs
[ 70%] Built target 3rdparty_lzf
[ 70%] Built target 3rdparty_jsoncpp
[ 70%] Built target 3rdparty_rply
[ 70%] Built target 3rdparty_qhull_r
[ 70%] Built target 3rdparty_qhullcpp
[ 70%] Built target Visualization
[ 70%] Built target Camera
[ 70%] Built target ColorMap
[ 70%] Built target Core
[ 70%] Built target Geometry
[ 70%] Built target Integration
[ 70%] Built target Odometry
[ 70%] Built target Registration
[ 70%] Built target Utility
[ 70%] Built target IO
[ 70%] Built target Open3D
[ 70%] Building CXX object src/Python/CMakeFiles/open3d_pybind.dir/open3d_pybind/core/device.cpp.o
`

**Comment by AndreV84** [comment]
the latter issue resolved with installing pip package;
another issue emerged while trying to process --make with recorded realsense images
`git_src/Open3D/theNded/Open3D/examples/Python/ReconstructionSystem$ python3 run_system.py /home/agx/o3d/git_src/Open3D/theNded/Open3D/examples/Cuda/ReconstructionSystem/config/intel/test.json --make
====================================
Configuration
====================================
                                    name : CMU - Test
                            path_dataset : /home/agx/o3d/git_src/Open3D/theNded/Open3D/examples/Python/ReconstructionSystem/dataset/realsense/
                          path_intrinsic : /home/agx/o3d/git_src/Open3D/theNded/Open3D/build/bin/examples/camera_intrinsic_D435i.json
                   n_frames_per_fragment : 100
                 n_keyframes_per_n_frame : 5
                               max_depth : 3.0
                              voxel_size : 0.05
                          max_depth_diff : 0.07
                            depth_factor : 1000.0
        preference_loop_closure_odometry : 0.1
    preference_loop_closure_registration : 10.0
                         tsdf_cubic_size : 3.0
                          depth_map_type : redwood
                               min_depth : 0.3
                              icp_method : color
                     global_registration : ransac
                  python_multi_threading : true
                         folder_fragment : fragments/
             template_fragment_posegraph : fragments/fragment_%03d.json
   template_fragment_posegraph_optimized : fragments/fragment_optimized_%03d.json
            template_fragment_pointcloud : fragments/fragment_%03d.ply
                            folder_scene : scene/
               template_global_posegraph : scene/global_registration.json
     template_global_posegraph_optimized : scene/global_registration_optimized.json
              template_refined_posegraph : scene/refined_registration.json
    template_refined_posegraph_optimized : scene/refined_registration_optimized.json
                    template_global_mesh : scene/integrated.ply
                    template_global_traj : scene/trajectory.log
                              debug_mode : False
OpenCV is detected. Using ORB + 5pt algorithm
making fragments from RGBD sequence.
Fragment 000 / 000 :: RGBD matching between frame : 0 and 1
Fragment 000 / 000 :: RGBD matching between frame : 0 and 5
Fragment 000 / 000 :: RGBD matching between frame : 0 and 10
Fragment 000 / 000 :: RGBD matching between frame : 0 and 15
Fragment 000 / 000 :: RGBD matching between frame : 0 and 20
Fragment 000 / 000 :: RGBD matching between frame : 1 and 2
Fragment 000 / 000 :: RGBD matching between frame : 2 and 3
Fragment 000 / 000 :: RGBD matching between frame : 3 and 4
Fragment 000 / 000 :: RGBD matching between frame : 4 and 5
Fragment 000 / 000 :: RGBD matching between frame : 5 and 6
Fragment 000 / 000 :: RGBD matching between frame : 5 and 10
Fragment 000 / 000 :: RGBD matching between frame : 5 and 15
Fragment 000 / 000 :: RGBD matching between frame : 5 and 20
Fragment 000 / 000 :: RGBD matching between frame : 6 and 7
Fragment 000 / 000 :: RGBD matching between frame : 7 and 8
Fragment 000 / 000 :: RGBD matching between frame : 8 and 9
Fragment 000 / 000 :: RGBD matching between frame : 9 and 10
Fragment 000 / 000 :: RGBD matching between frame : 10 and 11
Fragment 000 / 000 :: RGBD matching between frame : 10 and 15
Fragment 000 / 000 :: RGBD matching between frame : 10 and 20
Fragment 000 / 000 :: RGBD matching between frame : 11 and 12
Fragment 000 / 000 :: RGBD matching between frame : 12 and 13
Fragment 000 / 000 :: RGBD matching between frame : 13 and 14
Fragment 000 / 000 :: RGBD matching between frame : 14 and 15
Fragment 000 / 000 :: RGBD matching between frame : 15 and 16
Fragment 000 / 000 :: RGBD matching between frame : 15 and 20
Fragment 000 / 000 :: RGBD matching between frame : 16 and 17
Fragment 000 / 000 :: RGBD matching between frame : 17 and 18
Fragment 000 / 000 :: RGBD matching between frame : 18 and 19
Fragment 000 / 000 :: RGBD matching between frame : 19 and 20
Fragment 000 / 000 :: RGBD matching between frame : 20 and 21
Fragment 000 / 000 :: RGBD matching between frame : 21 and 22
Fragment 000 / 000 :: RGBD matching between frame : 22 and 23
Fragment 000 / 000 :: RGBD matching between frame : 23 and 24
[Open3D DEBUG] Validating PoseGraph - finished.
[Open3D DEBUG] [GlobalOptimizationLM] Optimizing PoseGraph having 25 nodes and 33 edges.
[Open3D DEBUG] Line process weight : 60.329928
[Open3D DEBUG] [Initial     ] residual : 1.152980e+04, lambda : 1.539617e+01
[Open3D DEBUG] [Iteration 00] residual : 2.189186e+02, valid edges : 6, time : 0.001 sec.
[Open3D DEBUG] [Iteration 01] residual : 1.814049e+02, valid edges : 6, time : 0.001 sec.
[Open3D DEBUG] [Iteration 02] residual : 1.813116e+02, valid edges : 6, time : 0.001 sec.
[Open3D DEBUG] Current_residual - new_residual < 1.000000e-06 * current_residual
[Open3D DEBUG] [GlobalOptimizationLM] total time : 0.004 sec.
[Open3D DEBUG] [GlobalOptimizationLM] Optimizing PoseGraph having 25 nodes and 30 edges.
[Open3D DEBUG] Line process weight : 63.495866
[Open3D DEBUG] [Initial     ] residual : 4.681930e+00, lambda : 2.223877e+01
[Open3D DEBUG] [Iteration 00] residual : 4.679709e+00, valid edges : 6, time : 0.001 sec.
[Open3D DEBUG] Current_residual - new_residual < 1.000000e-06 * current_residual
[Open3D DEBUG] [GlobalOptimizationLM] total time : 0.002 sec.
[Open3D DEBUG] CompensateReferencePoseGraphNode : reference : 0
Fragment 000 / 000 :: integrate rgbd frame 0 (1 of 25).
Traceback (most recent call last):
  File "run_system.py", line 68, in <module>
    make_fragments.run(config)
  File "/home/agx/o3d/git_src/Open3D/theNded/Open3D/examples/Python/ReconstructionSystem/make_fragments.py", line 183, in run
    for fragment_id in range(n_fragments))
  File "/home/agx/.local/lib/python3.6/site-packages/joblib/parallel.py", line 1029, in __call__
    if self.dispatch_one_batch(iterator):
  File "/home/agx/.local/lib/python3.6/site-packages/joblib/parallel.py", line 847, in dispatch_one_batch
    self._dispatch(tasks)
  File "/home/agx/.local/lib/python3.6/site-packages/joblib/parallel.py", line 765, in _dispatch
    job = self._backend.apply_async(batch, callback=cb)
  File "/home/agx/.local/lib/python3.6/site-packages/joblib/_parallel_backends.py", line 208, in apply_async
    result = ImmediateResult(func)
  File "/home/agx/.local/lib/python3.6/site-packages/joblib/_parallel_backends.py", line 572, in __init__
    self.results = batch()
  File "/home/agx/.local/lib/python3.6/site-packages/joblib/parallel.py", line 253, in __call__
    for func, args, kwargs in self.items]
  File "/home/agx/.local/lib/python3.6/site-packages/joblib/parallel.py", line 253, in <listcomp>
    for func, args, kwargs in self.items]
  File "/home/agx/o3d/git_src/Open3D/theNded/Open3D/examples/Python/ReconstructionSystem/make_fragments.py", line 165, in process_single_fragment
    intrinsic, config)
  File "/home/agx/o3d/git_src/Open3D/theNded/Open3D/examples/Python/ReconstructionSystem/make_fragments.py", line 139, in make_pointcloud_for_fragment
    intrinsic, config)
  File "/home/agx/o3d/git_src/Open3D/theNded/Open3D/examples/Python/ReconstructionSystem/make_fragments.py", line 127, in integrate_rgb_frames_for_fragment
    volume.integrate(rgbd, intrinsic, np.linalg.inv(pose))
RuntimeError: [Open3D ERROR] [ScalableTSDFVolume::Integrate] Unsupported image format.
`

**Comment by AndreV84** [comment]
`./RunSystem /home/user/o3d/git_src/Open3D/theNded/Open3D/examples/Cuda/ReconstructionSystem/config/intel/test.json
[Open3D INFO] Processing fragment 0 / 0
[Open3D INFO] RGBD Odometry between (0 1)
[Open3D INFO] RGBD Odometry between (1 2)
[Open3D INFO] RGBD Odometry between (2 3)
[Open3D INFO] RGBD Odometry between (3 4)
[Open3D INFO] RGBD Odometry between (4 5)
[Open3D INFO] RGBD Odometry between (5 6)
[Open3D INFO] RGBD Odometry between (6 7)
[Open3D INFO] RGBD Odometry between (7 8)
[Open3D INFO] RGBD Odometry between (8 9)
[Open3D INFO] RGBD Odometry between (9 10)
[Open3D INFO] RGBD Odometry between (10 11)
[Open3D INFO] RGBD Odometry between (11 12)
[Open3D INFO] RGBD Odometry between (12 13)
[Open3D INFO] RGBD Odometry between (13 14)
[Open3D INFO] RGBD Odometry between (14 15)
[Open3D INFO] RGBD Odometry between (15 16)
[Open3D INFO] RGBD Odometry between (16 17)
[Open3D INFO] RGBD Odometry between (17 18)
[Open3D INFO] RGBD Odometry between (18 19)
[Open3D INFO] RGBD Odometry between (19 20)
[Open3D INFO] RGBD Odometry between (20 21)
[Open3D INFO] RGBD Odometry between (21 22)
[Open3D INFO] RGBD Odometry between (22 23)
[Open3D INFO] RGBD Odometry between (23 24)
[Open3D INFO] Active subvolumes in volume: 389
[Open3D INFO] Active subvolumes in volume: 440
[Open3D INFO] Active subvolumes in volume: 460
[Open3D INFO] Active subvolumes in volume: 464
[Open3D INFO] Active subvolumes in volume: 473
[Open3D INFO] Active subvolumes in volume: 474
[Open3D INFO] Active subvolumes in volume: 485
[Open3D INFO] Active subvolumes in volume: 486
[Open3D INFO] Active subvolumes in volume: 500
[Open3D INFO] Active subvolumes in volume: 503
[Open3D INFO] Active subvolumes in volume: 509
[Open3D INFO] Active subvolumes in volume: 510
[Open3D INFO] Active subvolumes in volume: 512
[Open3D INFO] Active subvolumes in volume: 514
[Open3D INFO] Active subvolumes in volume: 515
[Open3D INFO] Active subvolumes in volume: 517
[Open3D INFO] Active subvolumes in volume: 521
[Open3D INFO] Active subvolumes in volume: 523
[Open3D INFO] Active subvolumes in volume: 524
[Open3D INFO] Active subvolumes in volume: 529
[Open3D INFO] Active subvolumes in volume: 530
[Open3D INFO] Active subvolumes in volume: 530
[Open3D INFO] Active subvolumes in volume: 534
[Open3D INFO] Active subvolumes in volume: 536
[Open3D INFO] Active subvolumes in volume: 541
[Open3D INFO] Writing 697 subvolumes.

CUDA error at /home/user/o3d/git_src/Open3D/theNded/Open3D/src/Cuda/Integration/ScalableMeshVolumeCudaKernel.cuh:117 code=701(cudaErrorLaunchOutOfResources) "cudaGetLastError()"`

**Comment by AndreV84** [comment]
another observed issue
 ./gdrive_download.sh 1dkA6Tjh-aEie1J8qGlY1NlyJVP0_FpBX
Downloading from 1dkA6Tjh-aEie1J8qGlY1NlyJVP0_FpBX...

** Resuming transfer from byte position 3103
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (33) HTTP server doesn't seem to support byte ranges. Cannot resume.

**Comment by yxlao** [comment]
@AndreV84 regarding the forked repo, we're not providing support here. Please post the issues in the corresponding repository.

BTW, ARM64 is now supported. See the build guide at https://github.com/intel-isl/Open3D/blob/master/docs/arm.rst. If you encounter any problem, please create a new issue.

**Comment by nanotuxi** [comment]
@yxlao Thanks for your hard work. Can confirm that the new gui is working fine on the jetson nano with cuda 10.2
